core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1109,10 +1109,12 @@ class ReplicaManager(val config: KafkaConfig,
         } catch {
           case e@ (_: UnknownTopicOrPartitionException |
                    _: NotLeaderOrFollowerException |
-                   _: OffsetOutOfRangeException |
                    _: PolicyViolationException |
                    _: KafkaStorageException) =>
             (topicPartition, LogDeleteRecordsResult(-1L, -1L, Some(e)))
+          case e: OffsetOutOfRangeException =>
+            debug("Error processing delete records operation on partition %s. %s".format(topicPartition, e.getMessage))
```

Comment:
```suggestion
            debug("Error processing delete records operation on partition %s".format(topicPartition), e)
```

You may want to log the exception as the last argument to get the complete stack trace

Reply:
This was deliberate, I figured that the error message is interesting since it describes why the error occurred, and which offsets were involved, but the stack trace probably isn't (sorry, it's been a while since I opened the PR, so the message on the mailing list was slightly off).

Do you prefer this with the stack trace, I'm happy to change it?

Reply:
It's very well a personal preference but I always prefer to have the stacktrace. The message might tell you the "why" but I've found the "where/how" to be immensely useful when debugging.

Reply:
Fair, committed your suggestion